File: SERVER/app/com/variant/server/boot/VariantConfig.scala
package com.variant.server.boot

import javax.inject.Inject
import play.api.Configuration
import com.variant.server.event.EventFlusherAppLogger

trait VariantConfigKey {
   val name: String
   val default: Any   
}

object VariantConfigKey {

   // Actual Keys
   lazy val EventFlusherClassName = new VariantConfigKey {
      val name = "variant.event.flusher.class.name"
      val default = classOf[EventFlusherAppLogger].getName
   }

   lazy val EventFlusherClassInit = new VariantConfigKey {
      val name = "variant.event.flusher.class.init"
      val default = "{}"
   }

   lazy val EventWriterPercentFull = new VariantConfigKey {
      val name = "event.writer.percent.full"
      val default = 50
   }

   lazy val EventWriterBufferSize = new VariantConfigKey {
      val name = "variant.event.writer.buffer.size"
      val default = 20000
   }

   lazy val EventWriterFlushMaxDelayMillis = new VariantConfigKey {
      val name = "variant.event.writer.flush.max.delay.millis"
      val default = 30000
   }

}
/**
 * Wrap Play provided applicaiton configuration in 
 * Variant specific adaptor.
 */
class VariantConfig @Inject() (playConfig: Configuration) {
  
   private def get(key: VariantConfigKey) = {
      playConfig.getString(key.name).getOrElse(key.default.toString())
   }
   
   def getString(key: VariantConfigKey) = get(key).asInstanceOf[String]
   def getInt(key: VariantConfigKey) = get(key).toInt
   def getLong(key: VariantConfigKey) = get(key).toLong
   
}

File: SERVER/app/com/variant/server/boot/VariantConfig.java
package com.variant.server.boot;

import play.api.Configuration;
import scala.Option;

/**
 * Wrap Play provided applicaiton configuration in 
 * Variant specific adaptor.
 */
public class VariantConfig {
  
	private final Configuration playConfig;
	
	VariantConfig(Configuration playConfig) {
		this.playConfig = playConfig;
	}
	
    private String get(VariantConfigKey key) {
    	Option<String> result = playConfig.getString(key.name, scala.Option.apply(null));
    	return (result.isEmpty() || result.get().isEmpty()) ? key.defaultValue.toString() : result.get();
    }

    public String getString(VariantConfigKey key) {
    	return get(key);
    }
    
    public int getInt(VariantConfigKey key) {
    	return Integer.parseInt(get(key));
    }

    public long getLong(VariantConfigKey key) {
    	return Long.parseLong(get(key));
    }

}
